### PR TITLE
#2552: blueprints page information hierarchy

### PR DESCRIPTION
**Pull request by mnholtz**
Changes the page layout of the Blueprints page by removing the standard Page header, and adding an "Explore" section to the sidebar.

- [x] Ideally would like to solve the vertical space/scrollbar for long lists of blueprints

### CODE REVIEW

**Comment by fregante** [comment]
Demo, it needs:

- some padding
- proper adjustments (e.g. `top: $variableOfTheHeaderHeight`)
- likely a `overflow:auto; max-height: 90vh`-ish on the sidebar just in case the sidebar is too tall. What I do usually is make sidebars stick if-and-only-if they fit in the viewport, but Facebook for example adds a scrollbar in that case.

https://user-images.githubusercontent.com/1402241/153539186-5c559148-a9d1-4fe9-83a1-25e6ad50e3f0.mov

**Comment by fregante** [comment]
Also the layout currently wraps/breaks at viewports less than 1200, likely due to a wide table

**Comment by twschiller** [comment]
> it takes ~11 seconds to switch layout when clicking these buttons on the "All blueprints" page

This is an issue with the RTK or useAsyncState. Tracking as a separate issue: https://github.com/pixiebrix/pixiebrix-extension/issues/2679

> Why not paginate this view like the rest of the options pages?

We're giving this a try first to see how it looks/feels to see if we like it. Infinite scroll is a better experience for an "explore" way of interacting, if you don't know what you already want from either your team market place, or eventually when we also layer in marketplace suggestions

**Comment by mnholtz** [comment]
Future work:
- [ ] Information hierarchy improvements to the Blueprint cards and table rows themselves (i.e. moving certain information to hover and click events)
- [ ] Apply appropriate vertical scrolling if the sidebar is too tall for the viewport
- [ ] Remove space between footer and Blueprints container to signal scrollable content
- [ ] Make the sidebar items more like "clickable" nav items

**Comment by fregante** [comment]
Not sure why my last commit was undone, this is now showing a triple scrollbar and a cut background fill. If virtual scrolling is the issue I'm sure it can be solved without boxing the content in a tiny viewport

https://user-images.githubusercontent.com/1402241/154035308-99a57468-d9eb-49ea-bcee-91821a22de2b.mov

**Comment by mnholtz** [comment]
> If virtual scrolling is the issue I'm sure it can be solved without boxing the content in a tiny viewport

I did go with AutoSizer to set up for virtualization. I'll have to find a way to do this alternatively to go back to the `position: sticky` method, although I think it's worth noting that I'm hoping to extend that bounding box to the bottom of the viewport at some point.

**Comment by twschiller** [comment]
> Not sure why my last commit was undone, this is now showing a triple scrollbar and a cut background fill. If virtual scrolling is the issue I'm sure it can be solved without boxing the content in a tiny viewport

We expect virtualization to be more straight-forward from a performance perspective. So we're going to go with the AutoSizer + Virtualization 

As Misha mentioned, we'll circle back to the footer/scrollbar issues in a future PR

We prioritized getting the performance issues resolved ASAP so people with early access can start testing the usability of the new page

File: src/options/pages/blueprints/BlueprintsCard.tsx [hunk 1]
```diff
@@ -187,19 +190,23 @@ const BlueprintsCard: React.FunctionComponent<{
...
       <ListFilters
         teamFilters={teamFilters}
         setGlobalFilter={setGlobalFilter}
       />
-      <Col>
-        <div className="d-flex justify-content-between align-items-center">
-          <h3 className="my-3">
+      <Col className={styles.mainContainer}>
+        <div
+          className={cx(
```
Comment (twschiller) [inline src/options/pages/blueprints/BlueprintsCard.tsx:200]: `cx` is not needed here since they're all constants. Just pass as a string

**Review by twschiller — APPROVED** [review]

**Pull request merged**